### PR TITLE
Add DnsZonePartner Parameter for New-AzureRmSqlManagedInstance cmdlet

### PR DESCRIPTION
**Pull request by ziwa-msft**
## Description
Add DnsZonePartner Parameter for New-AzureRmSqlManagedInstance cmdlet to support AutoDr for Managed Instance. Properties of the ManagedInstance model have also been updated to reflect new members DnsZone and DnsZonePartner

## Checklist

- [x] I have read the [_Submitting Changes_](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate [change log has been updated](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log)
- [x] The PR does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] the changes have gone through a [cmdlet design review](https://github.com/Azure/azure-powershell-cmdlet-review-pr)
    - [x] the cmdlet markdown files were [generated using the `platyPS` module](https://github.com/Azure/azure-powershell/blob/preview/documentation/development-docs/help-generation.md)

### CODE REVIEW

**Comment by ziwa-msft** [comment]
Link to .NetSDK: https://github.com/Azure/azure-sdk-for-net/pull/4801

**Comment by ziwa-msft** [comment]
I added the references to new ClientRuntime packages per advice from Maddie. I was seeing local failures due to package mismatch and reached out for help. She mentioned that there is a ClientRuntime upgrade ongoing and I should add these references for now to unblock testing.

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj [hunk 1]
```diff
@@ -61,6 +61,7 @@
     <Reference Include="Microsoft.Azure.Management.Network, Version=19.5.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Network.19.5.0-preview\lib\net452\Microsoft.Azure.Management.Network.dll</HintPath>
     </Reference>
+    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL" />
```
Comment (jaredmoo) [inline src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: I don't think that we should need to change Commands.Network or Commands.Profile . Were these changes intentional?

File: src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ManagedInstanceCrudScenarioTests.ps1 [hunk 2]
```diff
@@ -22,30 +22,31 @@ function Test-CreateManagedInstance
...
 	# Setup
 	$rg = Create-ResourceGroupForTest
-	 	
+	$rgName = "mi-wcus-demo-rg" 	
  	$managedInstanceName = Get-ManagedInstanceName
  	$version = "12.0"
  	$managedInstanceLogin = "dummylogin"
  	$managedInstancePassword = "Un53cuRE!"
- 	$subnetId = "/subscriptions/ee5ea899-0791-418f-9270-77cd8273794b/resourceGroups/cl_one/providers/Microsoft.Network/virtualNetworks/cl_initial/subnets/CooL"
+ 	$subnetId = "/subscriptions/a8c9a924-06c0-4bde-9788-e7b1370969e1/resourceGroups/mi-wcus-demo-rg/providers/Microsoft.Network/virtualNetworks/mi-wcus-vnet/subnets/mi-subnet-managed-instances"
```
Comment (jaredmoo) [inline src/ResourceManager/Sql/Commands.Sql.Test/ScenarioTests/ManagedInstanceCrudScenarioTests.ps1]: Since MI is in GA now, can you please change this test to be fully automated? This includes creating the virtual network. Check out SQL virtual network firewall rules for an example.

File: src/ResourceManager/Profile/Commands.Profile/Commands.Profile.csproj [hunk 3]
```diff
@@ -121,6 +121,18 @@
       <LastGenOutput>Resources.Designer.cs</LastGenOutput>
     </EmbeddedResource>
   </ItemGroup>
+  <ItemGroup>
```
Comment (cormacpayne) [inline src/ResourceManager/Profile/Commands.Profile/Commands.Profile.csproj]: @ziwa-msft please revert the change made in this file

File: src/ResourceManager/Profile/Commands.Profile/packages.config [hunk 4]
```diff
@@ -35,8 +35,8 @@
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net452" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net452" />
   <package id="Microsoft.PowerShell.5.ReferenceAssemblies" version="1.1.0" targetFramework="net452" />
-  <package id="Microsoft.Rest.ClientRuntime" version="2.3.13" targetFramework="net452" />
-  <package id="Microsoft.Rest.ClientRuntime.Azure" version="3.3.15" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.14" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Profile/Commands.Profile/packages.config]: @ziwa-msft please revert the change made in this file -- we have a separate PR open to update the version of ClientRuntime used throughout PowerShell

File: src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj [hunk 5]
```diff
@@ -53,12 +53,18 @@
...
     <Reference Include="Microsoft.Azure.Management.Sql, Version=1.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <SpecificVersion>False</SpecificVersion>
-      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Sql.1.20.0-preview\lib\net452\Microsoft.Azure.Management.Sql.dll</HintPath>
+      <HintPath>..\..\..\packages\Microsoft.Azure.Management.Sql.1.20.1-preview\lib\net452\Microsoft.Azure.Management.Sql.dll</HintPath>
       <Private>True</Private>
     </Reference>
     <Reference Include="Microsoft.Azure.Management.Storage">
       <HintPath>..\..\..\packages\Microsoft.Azure.Management.Storage.2.4.0-preview\lib\net40\Microsoft.Azure.Management.Storage.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.csproj]: @ziwa-msft please remove these references to ClientRuntime added in this file

File: src/ResourceManager/Sql/Commands.Sql.Test/packages.config [hunk 6]
```diff
@@ -17,6 +17,8 @@
   <package id="Microsoft.Bcl.Build" version="1.0.14" targetFramework="net45" />
   <package id="Microsoft.IdentityModel.Clients.ActiveDirectory" version="2.28.3" targetFramework="net45" />
   <package id="Microsoft.Net.Http" version="2.2.28" targetFramework="net45" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.14" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql.Test/packages.config]: @ziwa-msft same comment about removing the references to ClientRuntime in this file

File: src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj [hunk 7]
```diff
@@ -592,6 +590,12 @@
     <Reference Include="Microsoft.Data.Services.Client, Version=5.8.1.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
       <HintPath>..\..\..\packages\Microsoft.Data.Services.Client.5.8.2\lib\net40\Microsoft.Data.Services.Client.dll</HintPath>
     </Reference>
+    <Reference Include="Microsoft.Rest.ClientRuntime, Version=2.0.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/Commands.Sql.csproj]: @ziwa-msft same comment about removing the references to ClientRuntime in this file

File: src/ResourceManager/Sql/Commands.Sql/packages.config [hunk 8]
```diff
@@ -1,9 +1,11 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <packages>
-  <package id="Microsoft.Azure.Management.Sql" version="1.20.0-preview" targetFramework="net452" />
+  <package id="Microsoft.Azure.Management.Sql" version="1.20.1-preview" targetFramework="net452" />
   <package id="Microsoft.Azure.Management.Storage" version="2.4.0-preview" targetFramework="net45" />
   <package id="Microsoft.Data.Edm" version="5.8.2" targetFramework="net452" />
   <package id="Microsoft.Data.OData" version="5.8.2" targetFramework="net452" />
   <package id="Microsoft.Data.Services.Client" version="5.8.2" targetFramework="net452" />
+  <package id="Microsoft.Rest.ClientRuntime" version="2.3.14" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/packages.config]: @ziwa-msft same comment about removing the references to ClientRuntime in this file

File: src/ResourceManager/Sql/LegacySdk/SqlManagement.csproj [hunk 9]
```diff
@@ -50,5 +50,20 @@
       <Name>Commands.Profile</Name>
     </ProjectReference>
   </ItemGroup>
+  <ItemGroup>
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/LegacySdk/SqlManagement.csproj]: @ziwa-msft please revert the changes in this file -- you don't need to add a reference to ClientRuntime and I was under the impression since this was a generation of the old Sql SDK that you don't need a reference to the new Sql management SDK.

File: src/ResourceManager/Sql/LegacySdk/packages.config [hunk 10]
```diff
@@ -2,8 +2,11 @@
 <packages>
   <package id="Hyak.Common" version="1.0.3" targetFramework="net45" />
   <package id="Microsoft.Azure.Common" version="2.1.0" targetFramework="net45" />
+  <package id="Microsoft.Azure.Management.Sql" version="1.20.1-preview" targetFramework="net452" />
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/LegacySdk/packages.config]: @ziwa-msft same comment about reverting the changes made in this file

File: src/ResourceManager/Network/Commands.Network/Commands.Network.csproj (hunk 1 above)
Comment (ziwa-msft) [reply src/ResourceManager/Network/Commands.Network/Commands.Network.csproj]: it was auto-generated due to the new package version. reverting now.

**Comment by ziwa-msft** [comment]
@jaredmoo @cormacpayne  Please review and let me know if there is any other feedback

**Comment by cormacpayne** [comment]
PR to merge in the latest from preview: https://github.com/Azure/azure-powershell/pull/7574

**Comment by cormacpayne** [comment]
@ziwa-msft some build failures for you to look at:

### Jenkins

The `.json` files associated with the tests found in `ManagedInstanceCrudScenarioTests.cs` are not being copied to the output directory in the `.csproj` file.

### Travis

The version of the `Microsoft.Azure.Management.Sql` package was not updated in the [`Commands.Sql.Test.Netcore`](https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Sql/Commands.Sql.Test/Commands.Sql.Test.Netcore.csproj#L27) file.

**Comment by ziwa-msft** [comment]
@cormacpayne Hi Cormac. Looks like the default build failed because of CredScan issue. Any suggestion on how to proceed?

**Comment by ziwa-msft** [comment]
@cormacpayne  Looks like all checks have passed finally ;D

File: src/ResourceManager/Sql/Commands.Sql/ChangeLog.md [hunk 11]
```diff
@@ -21,6 +21,7 @@
 
 ## Version 4.11.5
 * Fixed issue where some backup cmdlets would not recognize the current azure subscription
+* Add DnsZonePartner Parameter for New-AzureRmSqlManagedInstance cmdlet to support AutoDr for Managed Instance.
```
Comment (cormacpayne) [inline src/ResourceManager/Sql/Commands.Sql/ChangeLog.md]: @ziwa-msft this should go under the `## Current Release` header, but we can move it when we do the versioning for the next release of this preview module
